packages/widgets-internal/swap/useUnmountingAnimation.ts

```diff
@@ -0,0 +1,80 @@
+import { RefObject, useEffect } from "react";
+
+/**
+ * Checks whether a given node is currently animating.
+ *
+ * @param node - The node to check for ongoing animations.
+ * @returns - true if the node is animating; false otherwise.
+ */
+function isAnimating(node?: Animatable | Document): boolean {
+  return (node?.getAnimations?.().length ?? 0) > 0;
+}
+
+/**
+ * This hook runs an unmounting animation on a specified node.
+ *
+ * The hook will also run the animation on any additional elements specified in
+ * the `animatedElements` parameter. If no additional elements are specified,
+ * the animation will only run on the provided node.
+ *
+ * After any of the animated elements have completed their animation, `node` is removed from its parent.
+ *
+ * @param node - The node to animate and remove.
+ * @param getAnimatingClass - A function that returns the CSS class to add to the animating elements.
+ * @param animatedElements - Additional elements to animate.
+ * @param skip - Whether to skip the animation and remove the node immediately.
+ */
+export function useUnmountingAnimation(
```

Comment:
seems like can be easily switching to framer instead?

Reply:
hmmm i dont know i like writing custom CSS anim props (although this specifc hook is borrowed from uniswap's impl, same to with some of there anims), however i always did prefer writing animations with styled components, than installing framer. reduce an extra dependancy wherever possible

Reply:
we have framer already